### PR TITLE
kubelet: Fix race when KillPod followed by IsPodPendingTermination

### LINKED ISSUE #98893: [Flaky] //pkg/kubelet TestSyncPodsDeletesWhenSourcesAreReadyPerQOS
**Which jobs are flaking**:

pull-kubernetes-bazel-test

**Which test(s) are flaking**:

TestSyncPodsDeletesWhenSourcesAreReadyPerQOS

**Testgrid link**:

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&include-filter-by-regex=%2F%2Fpkg%2Fkubelet:go

**Reason for failure**:

```
    kubelet_test.go:469: 
        	Error Trace:	kubelet_test.go:469
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
```

**Anything else we need to know**:

This seems to be the most flaky test in the testgrid.

/sig node

### PR DESCRIPTION
**Pull request by gjkim42**
**What type of PR is this?**

/kind bug
/kind failing-test

**What this PR does / why we need it**:

This PR fixes the race when `KillPod` followed by `IsPodPendingTerminationBy(UID|PodName)`.

When we called `KillPod`, the buffered channel received the request. Then `PerformPodKillingWork` fetched the request and added the given pod to the `terminationMap`. In the meantime, the caller could call `IsPodPendingTerminationByUID` which could return false because `terminationMap` had not yet been set.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/98893

**Special notes for your reviewer**:

Regarding `TestSyncPodsDeletesWhenSourcesAreReadyPerQOS` of https://github.com/kubernetes/kubernetes/issues/98893,
`HandlePodCleanup` calls `KillPod` followed by `cleanupOrphanedPodCgroups` which calls `IsPodPendingTerminationByUID` and the race occurs.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by gjkim42** [comment]
Could you please review this?
/cc @ehashman @rphillips @sjenning @wzshiming

**Comment by rphillips** [comment]
I have a fix for the bazel test over here https://github.com/kubernetes/kubernetes/pull/98938

**Comment by gjkim42** [comment]
```bash
12m25s: 1464 runs so far, 0 failures
15m55s: 1872 runs so far, 0 failures
```

**Comment by gjkim42** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/98923/pull-kubernetes-bazel-test/1359336998959058944

/test pull-kubernetes-bazel-test

File: pkg/kubelet/kubelet_test.go [hunk 1]
```diff
@@ -500,6 +501,39 @@ func TestSyncPodsDeletesWhenSourcesAreReady(t *testing.T) {
...
+	kl.podKiller.KillPod(&kubecontainer.PodPair{
+		APIPod:     nil,
+		RunningPod: pod,
+	})
+
+	if !kl.podKiller.IsPodPendingTerminationByUID(pod.ID) && len(fakeRuntime.KilledPods) == 0 {
+		t.Fatal("Race condition: When KillPod is complete, the pod should be pending termination or be killed")
+	}
+}
+
```
Comment (gjkim42) [inline pkg/kubelet/kubelet_test.go:536]: Added a unit test to simulate the race condition.

**Comment by gjkim42** [comment]
To make sure
/test pull-kubernetes-bazel-test

**Comment by gjkim42** [comment]
Tested with one cpu
```bash
stress ./kubelet.test -test.run TestSyncPodsDeletesWhenSourcesAreReadyPerQOS -test.v -test.cpu 1
...
7m10s: 840 runs so far, 36 failures (4.29%)
```

**Comment by gjkim42** [comment]
Closed in favor of https://github.com/kubernetes/kubernetes/pull/98938
Will make new PR to address the race condition of KillPod.

**Pull request closed without merging**